autowarefoundation/autoware.universe
# fix(compare_map_segmentation): update voxel_based for dynamic map loader with map grid coordinate

## Description

As [Map_loader server has provided map grid coordinates](https://github.com/autowarefoundation/autoware.universe/pull/3205), as updated of  https://github.com/autowarefoundation/autoware.universe/pull/3087 this PR uses new map_loader I/F for efficiently search map grid which input point cloud belong to and avoid the degradation of compare result for point on the map grid boundary. 

## Related links

[TIERIV COMPANY INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-27456)


## Tests performed

- Run autoware with split map pointcloud with pointcloud_map_metadata.yaml file.

- Confirm the processing time and voxel_based_compare_map_filter's output as the same as whole map pointcloud loading version. 

| -  |Whole map loading |Dynamic map loading|
| -  |-----------------|----------|
|Pointcloud number  | ![image](https://user-images.githubusercontent.com/94814556/230023395-92e2e96b-d6ad-4496-8e91-59e1356d91b7.png) | ![image](https://user-images.githubusercontent.com/94814556/230021143-a0a0859b-dd78-42c4-b1cd-1cc3bb4e126e.png)   |
|Processing time|  ![image](https://user-images.githubusercontent.com/94814556/230018034-9d2ad77f-ef4d-4bdb-b3a3-ce707c5b3758.png) | ![image](https://user-images.githubusercontent.com/94814556/230020826-0f15f9df-2d37-4326-924d-7f0ee3826aff.png)  |


## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
